lambdas/service/app.py

```diff
@@ -699,6 +702,40 @@ def validate_filters(filters):
             else:
                 raise BadRequestError(f'The relation in the `filters` parameter entry for `{facet}`'
                                       f' must be one of {valid_relations}')
+            if facet == 'organismAge':
+                validate_organism_age_filter(value)
+
+
+def validate_organism_age_filter(values):
+    """
+    >>> validate_organism_age_filter([{'value': '20', 'unit': 'year'}])
+
+    >>> validate_organism_age_filter([{'notval': '20', 'notu': 'year'}])
+    Traceback (most recent call last):
+    ...
+    chalice.app.BadRequestError: BadRequestError: A dictionary with entries for `value` and `unit` is required
+
+    >>> validate_organism_age_filter([{'value': '', 'unit': 'year'}])
+    Traceback (most recent call last):
+    ...
+    chalice.app.BadRequestError: BadRequestError: The `value` entry must not be empty
+
+    >>> validate_organism_age_filter([{'value': '20', 'unit': ''}])
+    Traceback (most recent call last):
+    ...
+    chalice.app.BadRequestError: BadRequestError: The `unit` entry must not be empty
+
+    >>> validate_organism_age_filter([{'value': None, 'unit': 'years'}])
+    Traceback (most recent call last):
+    ...
+    chalice.app.BadRequestError: BadRequestError: A `unit` entry must be accompanied by a `value` entry
```

Comment:
Why does BadRequestError occur twice?

Reply:
It seems like that is how [Chalice renders it](https://github.com/aws/chalice/blob/88ddf6790aa56603183d8d238805d3954c85baae/chalice/app.py#L109). A bug.